File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.ts
function getRow(rowIndex: number): number[] {
  if (rowIndex === 0) return [1]
  const ans: number[] = []
  const prevRow: number[] = getRow(rowIndex - 1)
  let l: number
  let r: number
  for (let i = 0; i <= prevRow.length; i++) {
    l = prevRow[i - 1] || 0
    r = prevRow[i] || 0
    ans.push(l + r)
  }
  return ans
}

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.rb
# @param {Integer} row_index
# @return {Integer[]}
def get_row(row_index)
  return [1] if row_index == 0
  ans = []
  prev_row = get_row(row_index-1)
  l = r = nil
  (0..prev_row.length).each do |i|
    l = i == 0 ? 0 : prev_row[i-1]
    r = i == prev_row.length ? 0 : prev_row[i]
    ans << l+r
  end
  ans
end